numbat/src/help.rs

```diff
@@ -0,0 +1,146 @@
+                        m::empty(),
+                        |accumulated_mk, single_line| {
+                            accumulated_mk
+                                + m::nl()
+                                + m::whitespace("  ")
+                                + single_line.clone()
+                                + m::nl()
+                        },
+                    ) + m::nl()
+                        + m::whitespace("    ")
+                        + m::operator("=")
+                        + m::space()
+                        + value.pretty_print()
+                        + type_
+                        + m::nl();
+                }
+                InterpreterResult::Continue => {
+                    full_output += statement_output.lock().unwrap().iter().fold(
+                        m::empty(),
+                        |accumulated_mk, single_line| {
+                            accumulated_mk
+                                + m::nl()
+                                + m::whitespace("  ")
+                                + single_line.clone()
+                                + m::nl()
+                        },
+                    );
+                }
+                InterpreterResult::Exit(_exit_status) => {
+                    println!("Interpretation Error.");
+                }
+            }
+        }
+        Err(NumbatError::ResolverError(e)) => {
+            context.print_diagnostic(e.clone());
+        }
+        Err(NumbatError::NameResolutionError(
+            e @ (NameResolutionError::IdentifierClash { .. }
+            | NameResolutionError::ReservedIdentifier(_)),
+        )) => {
+            context.print_diagnostic(e);
+        }
+        Err(NumbatError::TypeCheckError(e)) => {
+            context.print_diagnostic(e);
+        }
+        Err(NumbatError::RuntimeError(e)) => {
+            context.print_diagnostic(e);
+        }
+    }
+
+    full_output
+}
+
+pub fn help_markup(pretty_print: bool) -> m::Markup {
+    let mut output = m::nl()
+        + m::keyword("numbat")
+        + m::space()
+        + m::text(env!("CARGO_PKG_DESCRIPTION"))
+        + m::nl()
+        + m::text("You can start by trying one of the examples:")
+        + m::nl();
+
+    let examples = vec![
+        "8 km / (1 h + 25 min)",
+        "atan2(30 cm, 1 m) -> deg",
+        "let ω = 2 π c / 660 cm",
+        r#"print("Energy of red photons: {ℏ ω -> eV}")"#,
+    ];
+    let mut example_context = Context::new(BuiltinModuleImporter::default());
+    let _use_prelude_output = evaluate_example(&mut example_context, "use prelude", false);
+    for example in examples.iter() {
+        output += m::text(">>> ") + m::text(example) + m::nl();
+        output += evaluate_example(&mut example_context, example, pretty_print);
+        output += m::nl();
+    }
+    output += m::text("Full documentation:")
+        + m::space()
+        + m::keyword("https://numbat.dev/doc/")
+        + m::nl();
```

Comment:
Maybe let's move this up? "You can read the full documentation at …. For now, you can try one of these examples: …"